llm_studio/src/utils/modeling_utils.py

```diff
@@ -169,6 +183,79 @@ def load_checkpoint(
         logger.info(f"Weights loaded from: {weights_path}")
 
 
+def deepspeed_initialize(
+    model: torch.nn.Module,
+    optimizer: torch.optim.Optimizer,
+    lr_scheduler: torch.optim.lr_scheduler._LRScheduler,
+    training_data: torch.utils.data.Dataset,
+    validating_data: torch.utils.data.Dataset,
+    cfg: Any,
+):
+    mconfig = AutoConfig.from_pretrained(cfg.llm_backbone)
+    if hasattr(mconfig, "hidden_size"):
+        model_hidden_size = mconfig.hidden_size
+    elif hasattr(mconfig, "d_model"):
+        model_hidden_size = mconfig.d_model
+    else:
+        raise Exception(f"deepspeed do not support {cfg.llm_backbone}")
+    ds_config = {
+        "fp16": {
+            "enabled": True if cfg.architecture.backbone_dtype == "float16" else False,
+            "loss_scale_window": 100,
+        },
+        "bf16": {
+            "enabled": True if cfg.architecture.backbone_dtype == "bfloat16" else False,
+            "loss_scale_window": 100,
+        },
+        # https://www.deepspeed.ai/docs/config-json/#zero-optimizations-for-fp16-training
```

Comment:
we should probably throw a warning for int4 and int8 selections and force to float16

Reply:
[d1efef5](https://github.com/h2oai/h2o-llmstudio/pull/288/commits/d1efef56fd2a7de4de01dff4a1b511696294803c)

Raise error for better understanding they are using FP16 when deepspeed.